### ISSUE TITLE
schema-reader Error: Illuminate\Routing\UrlGenerator::__construct(): Argument #2 ($request) must be of type Illuminate\Http\Request

### ISSUE THREAD

**Issue by alexfleites** [VemtoOrg/vemto2-issues#92]
schema-reader Error: Illuminate\Routing\UrlGenerator::__construct(): Argument #2 ($request) must be of type Illuminate\Http\Request, null given, called in /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Routing/RoutingServiceProvider.php on line 64
schema-reader Error: #0 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Routing/RoutingServiceProvider.php(64): Illuminate\Routing\UrlGenerator->__construct(Object(Illuminate\Routing\RouteCollection), NULL, NULL)
#1 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(900): Illuminate\Routing\RoutingServiceProvider->Illuminate\Routing\{closure}(Object(Illuminate\Foundation\Application), Array)
#2 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(787): Illuminate\Container\Container->build(Object(Closure))
#3 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1026): Illuminate\Container\Container->resolve('url', Array, true)
#4 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(723): Illuminate\Foundation\Application->resolve('url', Array)
#5 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1011): Illuminate\Container\Container->make('url', Array)
#6 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(122): Illuminate\Foundation\Application->make('url', Array)
#7 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(149): app('url')
#8 /Users/MacBookPro/Sites/ado/boomtow/filament/app/Providers/Filament/AdminPanelProvider.php(53): asset('img/logo.svg')
#9 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/filament/filament/src/PanelProvider.php(15): App\Providers\Filament\AdminPanelProvider->panel(Object(Filament\Panel))
#10 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Collections/helpers.php(233): Filament\PanelProvider->Filament\{closure}()
#11 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/filament/filament/src/Facades/Filament.php(136): value(Object(Closure))
#12 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(1294): Filament\Facades\Filament::Filament\Facades\{closure}(Object(Filament\PanelRegistry), Object(Illuminate\Foundation\Application))
#13 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(1240): Illuminate\Container\Container->fireCallbackArray(Object(Filament\PanelRegistry), Array)
#14 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(805): Illuminate\Container\Container->fireResolvingCallbacks('Filament\\PanelR...', Object(Filament\PanelRegistry))
#15 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1026): Illuminate\Container\Container->resolve('Filament\\PanelR...', Array, true)
#16 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(723): Illuminate\Foundation\Application->resolve('Filament\\PanelR...', Array)
#17 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1011): Illuminate\Container\Container->make('Filament\\PanelR...', Array)
#18 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(122): Illuminate\Foundation\Application->make('Filament\\PanelR...', Array)
#19 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/filament/filament/src/FilamentManager.php(49): app('Filament\\PanelR...')
#20 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/filament/filament/src/FilamentServiceProvider.php(48): Filament\FilamentManager->__construct()
#21 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(900): Filament\FilamentServiceProvider->Filament\{closure}(Object(Illuminate\Foundation\Application), Array)
#22 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(787): Illuminate\Container\Container->build(Object(Closure))
#23 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1026): Illuminate\Container\Container->resolve('filament', Array, true)
#24 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(723): Illuminate\Foundation\Application->resolve('filament', Array)
#25 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1011): Illuminate\Container\Container->make('filament', Array)
#26 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(1446): Illuminate\Foundation\Application->make('filament')
#27 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(237): Illuminate\Container\Container->offsetGet('filament')
#28 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(208): Illuminate\Support\Facades\Facade::resolveFacadeInstance('filament')
#29 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(351): Illuminate\Support\Facades\Facade::getFacadeRoot()
#30 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/filament/filament/routes/web.php(13): Illuminate\Support\Facades\Facade::__callStatic('getPanels', Array)
#31 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Routing/Router.php(509): Illuminate\Support\ServiceProvider->{closure}(Object(Illuminate\Routing\Router))
#32 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Routing/Router.php(465): Illuminate\Routing\Router->loadRoutes(Object(Closure))
#33 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Routing/RouteRegistrar.php(194): Illuminate\Routing\Router->group(Array, Object(Closure))
#34 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/filament/filament/routes/web.php(12): Illuminate\Routing\RouteRegistrar->group(Object(Closure))
#35 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Support/ServiceProvider.php(159): require('/Users/MacBookP...')
#36 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/spatie/laravel-package-tools/src/PackageServiceProvider.php(152): Illuminate\Support\ServiceProvider->loadRoutesFrom('/Users/MacBookP...')
#37 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Spatie\LaravelPackageTools\PackageServiceProvider->boot()
#38 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#39 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure(Object(Closure))
#40 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#41 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#42 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1099): Illuminate\Container\Container->call(Array)
#43 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1080): Illuminate\Foundation\Application->bootProvider(Object(Filament\FilamentServiceProvider))
#44 [internal function]: Illuminate\Foundation\Application->Illuminate\Foundation\{closure}(Object(Filament\FilamentServiceProvider), 19)
#45 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1079): array_walk(Array, Object(Closure))
#46 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/BootProviders.php(17): Illuminate\Foundation\Application->boot()
#47 /Users/MacBookPro/Sites/ado/boomtow/filament/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(309): Illuminate\Foundation\Bootstrap\BootProviders->bootstrap(Object(Illuminate\Foundation\Application))
#48 phar:///Applications/Vemto.app/Contents/Resources/static/VMTTL2/index.php(31): Illuminate\Foundation\Application->bootstrapWith(Array)
#49 phar:///Applications/Vemto.app/Contents/Resources/static/VMTTL2/common/Vemto.php(111): {closure}()
#50 phar:///Applications/Vemto.app/Contents/Resources/static/VMTTL2/index.php(18): Vemto\Vemto::execute('schema-reader', Object(Closure))
#51 /Applications/Vemto.app/Contents/Resources/static/VMTTL2(15): require('phar:///Applica...')
#52 {main}

**Comment by base-zero** [comment]
I'm seeing this same error on my Laravel 11 application when trying to use Vemto for the first time this eveving.

**Comment by haringsrob** [comment]
Same here on laravel 11

**Comment by TiagoSilvaPereira** [comment]
@haringsrob @alexfleites @base-zero could you please give more details? 

- When does this problem happen?
- Is this a new project?
- Creating a new project from Vemto throws this error?
- Creating a new project manually and connecting it to Vemto throws this error?
- Does this problem happens on a project without Filament installed?
- How can we reproduce it?
- Would it be possible to send a video?

**Comment by base-zero** [comment]
Sure:

1.) When does this problem happen?
When trying to create a new project and connect directly to the database.

2.) Is this a new project?
Yes it's new project, as it was the first time trying the project out.

3.) Creating a new project from Vemto throws this error?
Yes, when trying to connect to the database the error box from Vemto pops up showing the same error as listed in this issue.

4.) Creating a new project manually and connecting it to Vemto throws this error?
Haven't tried setting thing up manually, I just used the app and got the error when trying to connect directly to the database.

5.) Does this problem happens on a project without Filament installed?
Yes tested with a Laravel Livewire app and get the same error about URL::generator

6.) How can we reproduce it?
Download the latest version of the app, and trying to connect to a Laravel 11 app running on PHP 8.3.

7.) Would it be possible to send a video?
If the above doesn't help I can try and create a video for you. Just let us know.

Hope that helps.

**Comment by TiagoSilvaPereira** [comment]
Hi @base-zero, thanks for your detailed response. I tried to reproduce it but without success. Maybe this was a temporary problem with Laravel? Could you please check if this is still happening?

**Comment by TiagoSilvaPereira** [comment]
Hi @base-zero is this still happening? Please check the following link: https://stackoverflow.com/questions/72520085/illuminate-routing-urlgenerator-construct-argument-2-request-must-be-o

**Comment by darkopavic** [comment]
I have solved the issue by removing asset() and url() functions from /app and /config folder

**Comment by TiagoSilvaPereira** [comment]
> I have solved the issue by removing asset() and url() functions from /app and /config folder

Thanks for reporting this! Were those functions generated automatically for a new Laravel project? Or did you add them manually?

**Comment by darkopavic** [comment]
I added them manually.

**Comment by MikeCraig418** [comment]
I have this same issue on my Filament projects.

When I remove the asset() function, the error is resolved.

```diff
class AdminPanelProvider extends PanelProvider
{
    public function panel(Panel $panel): Panel
    {
        return $panel
            ->default()
            ->id('admin')
            ->path('admin')
            ->login()
-           ->brandLogo(asset('images/f7logo.png'))
            ->colors([
                'primary' => Color::Amber,
            ])
            ->topNavigation()
```

**Comment by TiagoSilvaPereira** [comment]
@yo-mike seems this problem happens because Vemto starts the application on Console mode. I'll check what I can do to avoid this annoying issue

**Comment by TiagoSilvaPereira** [comment]
@yo-mike does your application fail when running some artisan command when you use this?

```php
->brandLogo(asset('images/f7logo.png'))
```

**Comment by MikeCraig418** [comment]
No -- it doesn't appear to have any impact on the console when the line is present.

I can provide you a repo link if you want.

Best Regards,

Mike

**Comment by TiagoSilvaPereira** [comment]
@yo-mike, thank you very much; I managed to fix this problem on Vemto 2.0.10 (you can use **asset()**, **url()** or **route()** helpers inside Filament code without problems).

----------

For other people using these helpers inside config files, it is recommended to remove them or use PHP_SAPI to check if you are in a CLI environment. This is not a Vemto limitation but a Laravel limitation. If you call these helpers from config files, Laravel can't boot properly in CLI environments, including when calling Artisan commands.

If Vemto finds this error now, it will generate the following Tip message:

> &#x1F535; it seems you are using <b>url(), asset() or route()</b> functions in your config files. Please remove them before trying to connect your application, or use PHP_SAPI to check if you are in a CLI environment. For example:
> 
>  'setting' => PHP_SAPI === 'cli' ? false : url('')

You can check this for more details:

https://stackoverflow.com/questions/72520085/illuminate-routing-urlgenerator-construct-argument-2-request-must-be-o
